src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -851,16 +857,34 @@ public void checkForWorkerSpecificRecipes()
                         }
                         if(allMatch)
                         {
-                            duplicateFound = true;
+                            duplicateFound = token;
                             break;
                         }
                     }
                 }
-                if(!duplicateFound)
+                if(duplicateFound == null)
                 {
                     addRecipeToList(recipeToken);    
                     colony.getRequestManager().onColonyUpdate(request -> request.getRequest() instanceof IDeliverable && ((IDeliverable) request.getRequest()).matches(recipeStorage.getPrimaryOutput()));
                 }
+                else if(newRecipe.isMustExist() && duplicateFound != recipeToken)
```

Comment:
isMustExist seems weird

Reply:
The name? or the functionality?

Reply:
the name

Reply:
Got it. Happy to 'break the standard' here of 'is' for boolean and switch to 'get'